Modules/Scripted/VPAWVisualize/VPAWVisualize.py

```diff
@@ -322,25 +336,135 @@ def onApplyButton(self):
             )
-
-            # Compute inverted output (if needed)
-            if self.ui.invertedOutputSelector.currentNode():
-                # If additional output volume is selected then result with inverted
-                # threshold is written there
-                self.logic.process(
-                    self.ui.inputSelector.currentNode(),
-                    self.ui.invertedOutputSelector.currentNode(),
-                    self.ui.imageThresholdSliderWidget.value,
-                    not self.ui.invertOutputCheckBox.checked,
-                    showResult=False,
+            # Display output
+            self.clearOutputWidgets()
+            self.addOutputWidgets(list_of_files)
+
+    def clearOutputWidgets(self):
+        slicer.mrmlScene.GetSubjectHierarchyNode().RemoveAllItems(True)
+        self.show_nodes = list()
+
+    def processP3(self, filename, properties):
+        with open(filename, "rb") as f:
+            contents = pk.load(f)
+
+        print(f"File type for {filename} is not currently supported")
+        print(f"{filename} contains {summary_repr(contents)}") # !!!
+        # !!! Create node from contents
+
+        return None
+
+    def createNode(self, filename, basename_repr, props):
+        # Determine the node type from the filename extension, using its
+        # immediate-ancestor directory's name if necessary.  Note: check for
+        # ".seg.nrrd" before checking for ".nrrd".
+        if filename.endswith(".seg.nrrd"):
+            node = slicer.util.loadSegmentation(filename, properties=props)
+            node.CreateClosedSurfaceRepresentation()
+        elif filename.endswith(".nrrd"):
+            directory = os.path.basename(os.path.dirname(filename))
+            if directory == "images":
+                node = slicer.util.loadVolume(filename, properties=props)
+                self.show_nodes.append(node)
+            elif directory == "segmentations_computed":
+                node = slicer.util.loadSegmentation(filename, properties=props)
+                node.CreateClosedSurfaceRepresentation()
+            else:
+                # Guess
+                node = slicer.util.loadVolume(filename, properties=props)
+        elif filename.endswith(".fcsv"):
+            node = slicer.util.loadMarkups(filename)
+        elif filename.endswith(".mha"):
+            node = slicer.util.loadVolume(filename, properties=props)
+        elif filename.endswith(".png"):
+            node = slicer.util.loadVolume(filename, properties=props)
+        elif filename.endswith(".p3"):
+            node = self.processP3(filename, properties=props)
+        elif filename.endswith(".xls"):
+            print(f"File type for {basename_repr} is not currently supported")
+            node = None
+        else:
+            print(f"File type for {basename_repr} is not recognized")
+            node = None
+        return node
+
+    def addOutputWidget(self, shNode, subject_item, filename):
```

Comment:
This shouldn't be called `addOutputWidget`. It's not adding widgets. It loads nodes and rearranges the subject hierarchy.

A docstring would be nice here too.

Reply:
Done.